docs/execution_providers/OpenVINO-ExecutionProvider.md

```diff
@@ -5,146 +5,121 @@ OpenVINO Execution Provider enables deep learning inference on Intel CPUs, Intel
 ## Build
 For build instructions, please see the [BUILD page](../../BUILD.md#openvino).
 
+## Dynamic device selection
+When ONNX Runtime is built with OpenVINO Execution Provider, a target hardware option needs to be provided. This build time option becomes the default target harware the EP schedules inference on. However, this target may be overriden at runtime to schedule inference on a different hardware as shown below.
+
+Note. This dynamic hardware selection is optional. The EP falls back to the build-time default selection if no dynamic hardware option value is specified.
+1. Python API
+```
+import onnxruntime
+onnxruntime.capi._pybind_state.set_openvino_device("<harware_option>")
+# Create session after this
+```
+2. C/C++ API
+```
+Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_OpenVINO(sf, "<hardware_option>"));
```

Comment:
1. <hardware_option> -- does that mean user still can choose CPU_FP32 if it is built for GPU_FP32?
2. If there are multiple VPU on the device, does it mean it will use them all? Is it possible to let user choose which VPU to use? For Nvidia CUDA, we use the second parameter as a device id so user can choose which card to use.

Reply:
1. That's correct. The underlying OpenVINO install package comes with components for all hardware targets. Hence this hardware_option can be set dynamically at runtime. The option chosen during build time will be the fallback option if no dynamic_option provided at run time. More details here: [OpenVINO-ExecutionProvider.md doc](https://github.com/intel/onnxruntime/blob/openvino-ep-v2/docs/execution_providers/OpenVINO-ExecutionProvider.md#dynamic-device-selection)

2. We currently don't have APIs down the stack that allow us to select a particular VPU device for inference, and hence we can't expose it upwards to ORT. This feature request is still in discussion phase internally.

Reply:
for your answer 2, once your guys make the final decision, please update the comments in openvino_provider_factory.h "\param device_id openvino **device id, starts from zero**."

Reply:
we will definitely need #2. the sooner the better. please keep us updated on the discussions. thanks!

Reply:
Will do.